docs/content/preview/drivers-orms/rust/yb-rust-postgres.md

```diff
@@ -0,0 +1,166 @@
+
+To start building your application, make sure you have met the [prerequisites](../#prerequisites).
+
+### Step 1: Set up the database connection
+
+The following table describes the connection parameters required to connect, including [smart driver parameters](../../smart-drivers/) for uniform and topology load balancing.
+
+| Parameter | Description | Default |
+| :-------- | :---------- | :------ |
+| host | Host name of the YugabyteDB instance. You can also enter [multiple addresses](#use-multiple-addresses). |  |
+| port | Listen port for YSQL | 5433 |
+| database/dbname | Database name |  |
+| user | User connecting to the database |  |
+| password | User password |  |
+| `load_balance` | [Uniform load balancing](../../smart-drivers/#cluster-aware-connection-load-balancing) | Defaults to upstream driver behavior unless set to 'true' |
+| `topology_keys` | [Topology-aware load balancing](../../smart-drivers/#topology-aware-connection-load-balancing) | If `load_balance` is true, uses uniform load balancing unless set to comma-separated geo-locations in the form `cloud.region.zone`. |
+| yb_servers_refresh_interval | If load_balance is true, the interval in seconds to refresh the servers list | 300 seconds|
+| fallback_to_topology_keys_only | When set to true, if all the servers in the primary and fallback placements are down, the smart driver does not fallback to the rest of the servers in the cluster, instead to the nodes given in the connection URL. | false |
+| failed_host_reconnect_delay_secs | Mark the server as "UP" only if `failed-host-reconnect-delay-secs` duration has elapsed from the last time it was marked "DOWN". | 5 seconds |
+
+The following is an example connection string for connecting to YugabyteDB:
+
+```sh
+postgresql://127.0.0.1:5433/yugabyte?user=yugabyte&password=yugabyte& \
+    yb_servers_refresh_interval=0& \
+    load_balance=true& \
+    topology_keys=cloud1.datacenter1.rack2
+```
+
+After the driver establishes the initial connection, it fetches the list of available servers from the cluster, and load-balances subsequent connection requests across these servers.
+
+#### Use multiple addresses
+
+You can specify multiple hosts in the connection string to provide alternative options during the initial connection in case the primary address fails.
+
+{{< tip title="Tip">}}
+To obtain a list of available hosts, you can connect to any cluster node and use the `yb_servers()` YSQL function.
+{{< /tip >}}
+
+Delimit the addresses using commas, as follows:
+
+```sh
+postgresql://127.0.0.1:5433,127.0.0.2:5433/yugabyte?user=yugabyte&password=yugabyte& \
+    yb_servers_refresh_interval=0& \
+    load_balance=true& \
+    topology_keys=cloud1.datacenter1.rack2&fallback_to_topology_keys_only=true
+```
+
+The hosts are only used during the initial connection attempt. If the first host is down when the driver is connecting, the driver attempts to connect to the next host in the string, and so on.
+
```

Comment:
Missing options to connect using TLS/SSL

Reply:
@HarshDaryani896 - what about SSL options?

Reply:
TLS/SSL steps added.